packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -1153,17 +1153,7 @@ class LocalizationsGenerator {
         case ST.placeholderExpr:
           assert(node.children[1].type == ST.identifier);
           final Node identifier = node.children[1];
-          // Check that placeholders exist.
-          final Placeholder? placeholder = message.placeholders[identifier.value];
-          if (placeholder == null) {
-            throw L10nParserException(
-              'Make sure that the specified placeholder is defined in your arb file.',
-              _inputFileNames[locale]!,
-              message.resourceId,
-              translationForMessage,
-              identifier.positionInMessage,
-            );
-          }
+          final Placeholder placeholder = message.placeholders[identifier.value]!;
```

Comment:
Do we know that the `!` will always find a value that is not null here and elsewhere in this file where it is being added?

Reply:
Yes! In the `Message` class where all the parsing and type inference for placeholders happen, we make sure that all types are correct and that all placeholders used in messages exist in the placeholders map.